src/Symfony/Component/Cache/Adapter/OpCacheAdapter.php

```diff
@@ -0,0 +1,366 @@
+            $value = unserialize($value);
+        }
+
+        $f = $this->createCacheItem;
+
+        return $f($key, $value, true);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getItems(array $keys = array())
+    {
+        if (null === $this->values) {
+            $this->initialize();
+        }
+
+        return $this->generateItems($keys);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function hasItem($key)
+    {
+        if (null === $this->values) {
+            $this->initialize();
+        }
+
+        return isset($this->values[$key]) || $this->fallbackPool->hasItem($key);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function clear()
+    {
+        if (null === $this->values) {
+            $this->initialize();
+        }
+
+        return $this->fallbackPool->clear() && !$this->values;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function deleteItem($key)
+    {
+        if (null === $this->values) {
+            $this->initialize();
+        }
+
+        return !isset($this->values[$key]) && $this->fallbackPool->deleteItem($key);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function deleteItems(array $keys)
+    {
+        if (null === $this->values) {
+            $this->initialize();
+        }
+
+        $deleted = true;
+        $fallbackKeys = array();
+
+        foreach ($keys as $key) {
+            if (isset($this->values[$key])) {
+                $deleted = false;
+            } else {
+                $fallbackKeys[] = $key;
+            }
+        }
+
+        if ($fallbackKeys) {
+            $deleted = $this->fallbackPool->deleteItems($fallbackKeys) && $deleted;
+        }
+
+        return $deleted;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function save(CacheItemInterface $item)
+    {
+        if (null === $this->values) {
+            $this->initialize();
+        }
+
+        return !isset($this->values[$item->getKey()]) && $this->fallbackPool->save($item);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function saveDeferred(CacheItemInterface $item)
+    {
+        if (null === $this->values) {
+            $this->initialize();
+        }
+
+        return !isset($this->values[$item->getKey()]) && $this->fallbackPool->saveDeferred($item);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function commit()
+    {
+        return $this->fallbackPool->commit();
+    }
+
+    /**
+     * Load the cache file.
+     */
+    private function initialize()
+    {
+        $this->values = @(include $this->file) ?: array();
+    }
+
+    /**
+     * Generator for items.
+     *
+     * @param array $keys
+     *
+     * @return \Generator
+     */
+    private function generateItems(array $keys)
+    {
+        $f = $this->createCacheItem;
+        $fallbackKeys = array();
+
+        foreach ($keys as $key) {
+            if (!is_string($key)) {
```

Comment:
indeed. Doing it in the generator will throw the exception too late (the iterator will throw it during iteration, while it must be throw by getItems)